src/Backend/FileOperations.vala

```diff
@@ -149,4 +150,39 @@ namespace Taxi {
             }
         }
     }
+
+    private int ask_overwrite (File destination) {
+        var message_dialog = new Granite.MessageDialog (
+            _("Replace existing file?"),
+            _("<i>\"%s\"</i> already exists. You can replace this file, replace all conflicting files or choose not to replace the file by skipping.".printf (destination.get_basename ())),
+            new ThemedIcon ("dialog-warning"),
+            Gtk.ButtonsType.CANCEL
+        ) {
+            modal = true
+        };
+
+        var replace_all_button = new Gtk.Button.with_label (_("Replace All Conflicts"));
+        message_dialog.add_action_widget (replace_all_button, ConflictFlag.REPLACE_ALL);
+
+        var skip_button = new Gtk.Button.with_label (_("Skip"));
+        message_dialog.add_action_widget (skip_button, ConflictFlag.SKIP);
+
+        var replace_button = new Gtk.Button.with_label (_("Replace"));
+        replace_button.add_css_class (Granite.STYLE_CLASS_SUGGESTED_ACTION);
+        message_dialog.add_action_widget (replace_button, ConflictFlag.REPLACE);
```

Comment:
`add_button` returns a Gtk.Widget so this is kind of an unnecessary refactor. For the one button that has a style class you can do:

```
        dialog.add_button (_("Replace All Conflicts"), ConflictFlag.REPLACE_ALL);
        dialog.add_button (_("Skip"), ConflictFlag.SKIP);
        var replace_button = dialog.add_button (_("Replace"), ConflictFlag.REPLACE);
        replace_button.add_css_class (Granite.STYLE_CLASS_SUGGESTED_ACTION);
```